### ISSUE TITLE
Don't create CNAME file with undefined value

### ISSUE THREAD

**Issue by jabby** [HubPress/hubpress.io#1]
When you edit your settings and don't add a CNAME value, there is a new CNAME file on the repository with undefined value inside. 

After that, for each build, github send a message : 
```
The page build completed successfully, but returned the following warning:

Bad CNAME format: undefined

For information on troubleshooting Jekyll see:

  https://help.github.com/articles/using-jekyll-with-pages#troubleshooting

If you have any questions please contact us at https://github.com/contact.
```

If you add a CNAME value and after remove this value, the file is deleted and a new save on settings don't create a new CNAME file.

**Comment by anthonny** [comment]
Fixed with the commit 5c6a4ae1882a7ba

But a better fix will be check the previous state of the settings and delete CNAME only if it's removed
Keep open

**Comment by anthonny** [comment]
I'll create an enhancement to refactor CNAME function
